src/containers/Tabs/Account.tsx

```diff
@@ -49,73 +47,59 @@ const AccountTabContainer: React.FC<AccountTabContainerProps> = ({ isOpen }) =>
     openToast
   } = useTab(whitelist, (identifier: string) => ({ address: identifier }));
 
-  const { drizzle } = drizzleReactHooks.useDrizzle();
-
-  const { addAccount, removeAccount } = drizzle.contracts.AccountRules.methods as {
-    addAccount: (value: string) => TransactionObject<never>;
-    removeAccount: (value: string) => TransactionObject<never>;
-  };
-
   const handleAdd = async (value: string) => {
-    const gasLimit = await addAccount(value).estimateGas({
-      from: userAddress
-    });
-    addAccount(value)
-      .send({ from: userAddress, gas: gasLimit * 4 })
-      .on('transactionHash', () => {
-        toggleModal('add')();
-        addTransaction(value, PENDING_ADDITION);
-      })
-      .on('receipt', receipt => {
-        const event = idx(receipt, _ => _.events.AccountAdded);
-        const added = Boolean(idx(event, _ => _.returnValues.accountAdded));
-        if (!event) {
+    // const gas = await accountRulesContract!.estimate.addAccount(value)
+    try {
+      const tx = await accountRulesContract!.functions.addAccount(value);
+      toggleModal('add')();
+      addTransaction(value, PENDING_ADDITION);
+      const receipt = await tx.wait(1); // wait on receipt confirmations
+      const addEvent = receipt.events!.filter(e => e.event && e.event === 'AccountAdded').pop();
+      if (!addEvent) {
+        openToast(value, FAIL, `Error while processing account: ${value}`);
+      } else {
+        const addSuccessResult = idx(addEvent, _ => _.args[1]);
```

Comment:
@Errorific is `_.args[1]` meant to be 0 or 1?

Reply:
1, in the account case its 
```
event AccountAdded(
        address account,
        bool accountAdded
);
```

in the enode case its 
```
    event NodeAdded(
        bool nodeAdded
    );
```

Reply:
got it!